### PR TITLE
wire up meta client shard methods

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: services/meta/client.go [hunk 1]
```diff
@@ -511,24 +512,111 @@ func (c *Client) UserCount() int {
 	return len(c.data.Users)
 }
 
+// ShardGroupsByTimeRange returns a list of all shard groups on a database and policy that may contain data
+// for the specified time range. Shard groups are sorted by start time.
 func (c *Client) ShardGroupsByTimeRange(database, policy string, min, max time.Time) (a []ShardGroupInfo, err error) {
-	return nil, nil
+	// Find retention policy.
```
Comment (pauldix) [inline services/meta/client.go]: you'll need an RLock

Comment (pauldix) [inline services/meta/client.go]: you just need it to grab a reference to `data`

File: services/meta/client.go [hunk 2]
```diff
@@ -511,24 +512,111 @@ func (c *Client) UserCount() int {
...
+
+	return c.retryUntilExec(internal.Command_DeleteShardGroupCommand, internal.E_DeleteShardGroupCommand_Command, cmd)
 }
 
+// PrecreateShardGroups creates shard groups whose endtime is before the 'to' time passed in, but
+// is yet to expire before 'from'. This is to avoid the need for these shards to be created when data
+// for the corresponding time range arrives. Shard creation involves Raft consensus, and precreation
+// avoids taking the hit at write-time.
 func (c *Client) PrecreateShardGroups(from, to time.Time) error {
+	for _, di := range c.data.Databases {
```
Comment (pauldix) [inline services/meta/client.go]: rlock to grab a reference to data

File: services/meta/client.go [hunk 3]
```diff
@@ -511,24 +512,111 @@ func (c *Client) UserCount() int {
...
+		return nil
+	}
 	return nil
 }
 
-func (c *Client) ShardOwner(shardID uint64) (string, string, *ShardGroupInfo) {
-	return "", "", nil
+// ShardOwner returns the owning shard group info for a specific shard.
+func (c *Client) ShardOwner(shardID uint64) (database, policy string, sgi *ShardGroupInfo) {
+	for _, dbi := range c.data.Databases {
```
Comment (pauldix) [inline services/meta/client.go]: rlock for reference to data

**Comment by pauldix** [comment]
few comments on locks, otherwise +1

File: services/meta/client.go (hunk 1 above)
Comment (dgnorton) [inline services/meta/client.go]: :facepalm:

**Comment by corylanou** [comment]
+1

**Pull request merged**